### PR TITLE
feat(web): add test for workflow and step

### PR DESCRIPTION
**Pull request by davidsoderberg**
Screenshot 2024-06-19 at 13 50 26

### CODE REVIEW

File: apps/web/src/pages/templates/editor_v2/TemplateStepEditorV2.tsx [hunk 1]
```diff
@@ -37,13 +41,29 @@ export const WorkflowsStepEditorPageV2 = () => {
...
+        subject: '',
+        payload,
+        inputs,
+        to: currentUser?.email || '',
+        bridge: true,
+        content: '',
+      });
+      successMessage('Test sent successfully!');
+    } catch (e: any) {
+      errorMessage(e.message || 'Un-expected error occurred');
```
Comment (SokratisVidros) [inline apps/web/src/pages/templates/editor_v2/TemplateStepEditorV2.tsx]:
```suggestion
      errorMessage(e.message || 'Unexpected error occurred');
```

File: apps/web/src/pages/templates/editor_v2/TemplateStepEditorV2.tsx [hunk 2]
```diff
@@ -37,13 +41,29 @@ export const WorkflowsStepEditorPageV2 = () => {
...
+        workflowId: (workflow as any)?.rawData?.workflowId,
+        contentType: 'customHtml',
+        subject: '',
+        payload,
+        inputs,
+        to: currentUser?.email || '',
+        bridge: true,
+        content: '',
+      });
+      successMessage('Test sent successfully!');
```
Comment (SokratisVidros) [inline apps/web/src/pages/templates/editor_v2/TemplateStepEditorV2.tsx]: The copywriting in the `successMessage` needs to be more accurate as we didn't send a test.
```suggestion
      successMessage('Test run has started!');
```

File: apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx [hunk 3]
```diff
@@ -29,11 +33,27 @@ export const WorkflowsStepEditorPage = () => {
...
-  const handleTestClick = () => {
-    // TODO: this is just a temporary step for connecting the prototype
-    navigate(ROUTES.STUDIO_FLOWS_TEST_STEP);
+  const { mutateAsync: testSendEmailEvent, isLoading: isTestingEmail } = useMutation(testSendEmailMessage);
+
+  const handleTestClick = async () => {
+    try {
+      await testSendEmailEvent({
+        stepId,
+        workflowId: workflow.workflowid,
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx]: Why do we have this discrepancy with casing on the `id`? We should eliminate it.

```suggestion
        workflowId: workflow.workflowId,
```

File: apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx [hunk 4]
```diff
@@ -29,11 +33,27 @@ export const WorkflowsStepEditorPage = () => {
...
+        contentType: 'customHtml',
+        subject: '',
+        payload,
+        inputs,
+        to: currentUser?.email || '',
+        bridge: true,
+        content: '',
+      });
+      successMessage('Test sent successfully!');
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx]: Ditto.

File: apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx [hunk 5]
```diff
@@ -29,11 +33,27 @@ export const WorkflowsStepEditorPage = () => {
...
+    } catch (e: any) {
+      errorMessage(e.message || 'Un-expected error occurred');
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx]:
```suggestion
      errorMessage(e.message || 'Unexpected error occurred');
```

File: apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        </HStack>
+        <Stack gap="margins.layout.Input.input-input">
+          <Controller
+            control={control}
+            name={`subscriberId`}
+            render={({ field }) => {
+              return <Input {...field} label="Subscriber ID" />;
+            }}
+          />
+          <When truthy={stepTypes.includes(ChannelTypeEnum.EMAIL)}>
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx:58]: 🙃 I am not a big fan of this `When` component, as it is more verbose than a boolean JSX expression. It has widespread usage in the codebase but i question it's inner DX value.

```suggestion
{stepTypes.includes(ChannelTypeEnum.EMAIL) && <SomeJSX />}
```

File: apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            <Controller
+              control={control}
+              name={`email`}
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx]:
```suggestion
              name="email"
```

File: apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+              render={({ field }) => {
+                return <Input {...field} label="Email" />;
+              }}
+            />
+          </When>
+          <When truthy={stepTypes.includes(ChannelTypeEnum.SMS)}>
+            <Controller
+              control={control}
+              name={`phone`}
+              render={({ field }) => {
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx]:
```suggestion
              render={({ field }) => <Input {...field} label="Phone number" />
```

File: apps/web/src/studio/components/workflows/test-workflow/WorkflowsTestPage.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  const onChange = (payloadValues, toValues) => {
+    if (toValues) {
+      setTo(toValues);
+    }
+    if (payloadValues) {
+      setPayload(payloadValues);
+    }
+  };
+
+  if (isAuthLoading || isWorkflowLoading || isTestLoading || !currentUser) {
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/test-workflow/WorkflowsTestPage.tsx]: `isAuthLoading` ensures that user and org is loaded. 

```suggestion
  if (isAuthLoading || isWorkflowLoading || isTestLoading) {
```

File: apps/web/src/studio/components/workflows/test-workflow/WorkflowsTestPage.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    return null;
+  }
+
+  return (
+    <WorkflowsPageTemplate
+      title="Test workflow"
+      description="Test trigger as if you sent it from your API"
```
Comment (SokratisVidros) [inline apps/web/src/studio/components/workflows/test-workflow/WorkflowsTestPage.tsx]:
```suggestion
      description="Trigger a test run for this workflow"
```

**Review by SokratisVidros — APPROVED** [review]

**Comment by SokratisVidros** [comment]
@davidsoderberg I noticed in the screenshot that the payload inputs are not aligned with the rest of the form. There seems to be an extra left padding. Can we please fix that?

**Comment by davidsoderberg** [comment]
> @davidsoderberg I noticed in the screenshot that the payload inputs are not aligned with the rest of the form. There seems to be an extra left padding. Can we please fix that?

I will check with @ainouzgali as it is something in the json schema form component that creates that padding

**Pull request merged**